### PR TITLE
Some block mappings

### PR DESCRIPTION
**Pull request by YanisBft**
Some block mappings, including most of the new methods in `Blocks`.

### CODE REVIEW

**Review by Juuxel — APPROVED** [review]

File: mappings/net/minecraft/block/Blocks.mapping [hunk 1]
```diff
@@ -1,4 +1,26 @@
 CLASS net/minecraft/class_2246 net/minecraft/block/Blocks
+	METHOD method_26106 getLeavesSettings ()Lnet/minecraft/class_2397;
+	METHOD method_26107 getLightLevelFromBlockState (I)Ljava/util/function/ToIntFunction;
+		ARG 0 level
```
Comment (liach) [inline mappings/net/minecraft/block/Blocks.mapping]: Maybe 
```suggestion
		ARG 0 litLevel
```
as the result function only returns this level when the block state has "lit" property's value equal to "true"

File: mappings/net/minecraft/block/Blocks.mapping [hunk 2]
```diff
@@ -1,4 +1,26 @@
...
+		ARG 0 color
+	METHOD method_26110 getShulkerBoxSettings (Lnet/minecraft/class_1767;Lnet/minecraft/class_4970$class_2251;)Lnet/minecraft/class_2480;
+		ARG 0 color
+		ARG 1 settings
+	METHOD method_26115 getStemSettings (Lnet/minecraft/class_3620;)Lnet/minecraft/class_2248;
+		ARG 0 materialColor
+	METHOD method_26117 getLogSettings (Lnet/minecraft/class_3620;Lnet/minecraft/class_3620;)Lnet/minecraft/class_2465;
+		ARG 0 topMaterialColor
+		ARG 1 sideMaterialColor
+	METHOD method_26119 getPistonSettings (Z)Lnet/minecraft/class_2665;
```
Comment (liach) [inline mappings/net/minecraft/block/Blocks.mapping]: `createPiston(Block)`, as this returns a ready-to-go `PistonBlock` than just a `Block/AbstractBlock.Settings`

File: mappings/net/minecraft/block/Blocks.mapping [hunk 3]
```diff
@@ -1,4 +1,26 @@
...
+	METHOD method_26107 getLightLevelFromBlockState (I)Ljava/util/function/ToIntFunction;
+		ARG 0 level
+	METHOD method_26109 getBedSettings (Lnet/minecraft/class_1767;)Lnet/minecraft/class_2244;
+		ARG 0 color
+	METHOD method_26110 getShulkerBoxSettings (Lnet/minecraft/class_1767;Lnet/minecraft/class_4970$class_2251;)Lnet/minecraft/class_2480;
+		ARG 0 color
+		ARG 1 settings
+	METHOD method_26115 getStemSettings (Lnet/minecraft/class_3620;)Lnet/minecraft/class_2248;
+		ARG 0 materialColor
+	METHOD method_26117 getLogSettings (Lnet/minecraft/class_3620;Lnet/minecraft/class_3620;)Lnet/minecraft/class_2465;
```
Comment (liach) [inline mappings/net/minecraft/block/Blocks.mapping]: Also `createLog(Block)`, same reasoning as piston, returns a ready-to-go `PillarBlock`

File: mappings/net/minecraft/block/Blocks.mapping [hunk 4]
```diff
@@ -1,4 +1,26 @@
 CLASS net/minecraft/class_2246 net/minecraft/block/Blocks
+	METHOD method_26106 getLeavesSettings ()Lnet/minecraft/class_2397;
+	METHOD method_26107 getLightLevelFromBlockState (I)Ljava/util/function/ToIntFunction;
+		ARG 0 level
+	METHOD method_26109 getBedSettings (Lnet/minecraft/class_1767;)Lnet/minecraft/class_2244;
+		ARG 0 color
+	METHOD method_26110 getShulkerBoxSettings (Lnet/minecraft/class_1767;Lnet/minecraft/class_4970$class_2251;)Lnet/minecraft/class_2480;
+		ARG 0 color
+		ARG 1 settings
+	METHOD method_26115 getStemSettings (Lnet/minecraft/class_3620;)Lnet/minecraft/class_2248;
```
Comment (liach) [inline mappings/net/minecraft/block/Blocks.mapping]: Same, `create(Nether)Stem(Block)` (the sound group refers to "nether stem")

File: mappings/net/minecraft/block/Blocks.mapping [hunk 5]
```diff
@@ -1,4 +1,26 @@
 CLASS net/minecraft/class_2246 net/minecraft/block/Blocks
+	METHOD method_26106 getLeavesSettings ()Lnet/minecraft/class_2397;
+	METHOD method_26107 getLightLevelFromBlockState (I)Ljava/util/function/ToIntFunction;
```
Comment (liach) [inline mappings/net/minecraft/block/Blocks.mapping]:
```suggestion
	METHOD method_26107 createLitBasedLightLevelGetter (I)Ljava/util/function/ToIntFunction;
```
Maybe need a more elegant name for this, as this does
```java
      return (state) -> {
         return state.get(Properties.LIT) ? parameter : 0;
      };
```

File: mappings/net/minecraft/block/Blocks.mapping [hunk 6]
```diff
@@ -1,4 +1,26 @@
 CLASS net/minecraft/class_2246 net/minecraft/block/Blocks
+	METHOD method_26106 getLeavesSettings ()Lnet/minecraft/class_2397;
```
Comment (liach) [inline mappings/net/minecraft/block/Blocks.mapping]: Same as piston, `createLeaves(Block)`

File: mappings/net/minecraft/block/Blocks.mapping [hunk 7]
```diff
@@ -1,4 +1,26 @@
...
+	METHOD method_26107 getLightLevelFromBlockState (I)Ljava/util/function/ToIntFunction;
+		ARG 0 level
+	METHOD method_26109 getBedSettings (Lnet/minecraft/class_1767;)Lnet/minecraft/class_2244;
```
Comment (liach) [inline mappings/net/minecraft/block/Blocks.mapping]: `createBed(Block)`, same as piston

File: mappings/net/minecraft/block/Blocks.mapping [hunk 8]
```diff
@@ -1,4 +1,26 @@
...
+		ARG 0 color
+	METHOD method_26110 getShulkerBoxSettings (Lnet/minecraft/class_1767;Lnet/minecraft/class_4970$class_2251;)Lnet/minecraft/class_2480;
```
Comment (liach) [inline mappings/net/minecraft/block/Blocks.mapping]: `createShulkerBox(Block)`, same as piston

File: mappings/net/minecraft/block/Blocks.mapping [hunk 9]
```diff
@@ -1,4 +1,26 @@
...
+		ARG 0 color
+		ARG 1 settings
+	METHOD method_26115 getStemSettings (Lnet/minecraft/class_3620;)Lnet/minecraft/class_2248;
+		ARG 0 materialColor
+	METHOD method_26117 getLogSettings (Lnet/minecraft/class_3620;Lnet/minecraft/class_3620;)Lnet/minecraft/class_2465;
+		ARG 0 topMaterialColor
+		ARG 1 sideMaterialColor
+	METHOD method_26119 getPistonSettings (Z)Lnet/minecraft/class_2665;
+		ARG 0 sticky
+	METHOD method_26120 getStainedGlassSettings (Lnet/minecraft/class_1767;)Lnet/minecraft/class_2506;
```
Comment (liach) [inline mappings/net/minecraft/block/Blocks.mapping]: Same `createStainedGlass(Block)` as piston

**Review by liach — CHANGES_REQUESTED** [review]
Besides these, you should name `method_26113` and `method_26122` as well, which means like "always pass/return true" and "always fail/return false" as they are used as method reference for where `AbstractBlock.ContextPredicate` is needed

**Review by liach — APPROVED** [review]
Nice!

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**